discovery/spec.md

```diff
@@ -917,108 +931,111 @@ Content-Type: application/json
-than the `epoch` of any existing Service of the same `id` or the request MUST
-fail.
-
-The follow responses are defined by this specification:
-- `200 OK` if Service collection was updated.
-  - The HTTP Response Body MUST include the Service values resulting from the
-    successful processing of the request. The response MAY include attributes
-    and values which were added by the Discovery Endpoint.
+discovery endpoint MUST assign a value that is greater than the existing
+`epoch` value for the Service.
+
+If an `epoch` is given in the request and it is not greater than the current
+Service's value, then the request MUST fail.
+
+Aside from `id` and `epoch`, unless there is an out of band agreement,
+all mandatory attributes MUST be present in the request, and a Discovery
+Endpoint MUST reject requests that are missing such attributes.
+
+The Discovery Endpoint MUST ignore attempts to modify read-only or immutable
+attributes - such as `id` and `url`.
+
+Upon successfully processing the request, the HTTP response body MUST be
+the Service values resulting from the successful processing of the request.
+The response SHOULD include the current values for all attributes, even if
+the Discovery Endpoint added or modified some value during the processing
+of the request.
+
+The following responses are defined by this specification:
+- `200 OK` if specified Service was updated successfullly.
 - `400 Bad Request` if the `id` in the path and body are not the same or some
-  other constraint failure is found
-- `404 Not Found` if there is no Service with the specified `id`
-- `409 Conflict` if the given `epoch` was less than or equal to the existing
-  `epoch` of the Service with the given `id`
+  other constraint failure is found.
+- `409 Conflict` if the given `epoch` is not greater than the current value.
 
 Other responses are allowed, but not defined by this specification.
 
-The format of the HTTP Request MUST adhere to the following:
+The format of the HTTP request MUST adhere to the following:
 ```
 Content-Type: application/json
 
 {
   "url": "{url}",
-  "epoch": "{id}",
   "name": "{name}",
   ... remainder of Service attributes ...
 }
 ```
 
-The format of the `200 OK` HTTP Response MUST adhere to the following:
+The format of the `200 OK` HTTP response MUST adhere to the following:
 ```
 200 OK
 Content-Type: application/json
 
 {
   "id": "{id}",
-  "epoch": "{id}",
+  "epoch": "{epoch}",
   ... remainder of Service attributes ...
 }
 ```
 
-#### `DELETE /services/{id}`
+#### `DELETE /services/{id}[?epoch={epoch}]`
+
+This MUST delete the Service as identified by the `id` in the URL. If there
+is no Service with the specified `id` then it is considered to have already
+been deleted and the request will have no change to the Discovery Endpoint.
+A missing Service is not considered an error condition.
 
-This MUST delete the Service at the referenced URL.
+This specification does not define a Body for the request message and the
```

Comment:
This PR lower-cases Response and Request - maybe do the same here for Body?

Reply:
done